tuxedo-core/src/types.rs

```diff
@@ -148,6 +155,46 @@ pub struct Output<V> {
     pub verifier: V,
 }
 
+/// Similar to Substrate's ValidTransaction. This type indicates that a transaction
+/// has passed the standard UTXO checks and is either valid now or may be in the future.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Debug, PartialEq, Eq, Clone, TypeInfo)]
+pub struct PreliminarilyValidTransaction<A> {
+    pub provides: Vec<Vec<u8>>,
+    pub requires: Vec<Vec<u8>>,
```

Comment:
any better description for what these` Vec<Vec<u8>>` are? An alias of any kind? if not is what it is.